packages/proxy/test/unit/http/util/service-worker-manager.spec.ts

```diff
@@ -1,621 +1,855 @@
-      requestId: 'id-4',
-      method: 'GET',
-      url: 'http://example.com/quux.js',
-      headers: {},
-      resourceType: 'fetch',
-      originalResourceType: undefined,
-      initiator: {
-        type: 'script',
-        stack: {
-          callFrames: [{
-            url: 'http://example.com/bar.js?foo=bar',
-            lineNumber: 1,
-            columnNumber: 1,
-            functionName: '',
-            scriptId: '1',
+        })
+
+        // update the service worker to be in a non-activated state
+        manager.updateServiceWorkerVersions({
+          versions: [{
+            versionId: '1',
+            runningStatus: 'running',
+            registrationId: '2',
+            scriptURL: 'http://localhost:8080/sw.js',
+            status: 'activating',
           }],
-        },
-      },
-      documentURL: 'http://localhost:8080/index.html?foo=bar',
-      cdpRequestWillBeSentTimestamp: 0,
-      cdpRequestWillBeSentReceivedTimestamp: 0,
-    })).to.be.false
-
-    // A script request emanated from a different css request is not controlled
-    expect(manager.processBrowserPreRequest({
-      requestId: 'id-5',
-      method: 'GET',
-      url: 'http://example.com/quux.css',
-      headers: {},
-      resourceType: 'fetch',
-      originalResourceType: undefined,
-      initiator: {
-        type: 'script',
-        url: 'http://example.com/baz.css?foo=bar',
-      },
-      documentURL: 'http://localhost:8080/index.html?foo=bar',
-      cdpRequestWillBeSentTimestamp: 0,
-      cdpRequestWillBeSentReceivedTimestamp: 0,
-    })).to.be.false
-
-    // A script request emanated from a different document is not controlled
-    expect(manager.processBrowserPreRequest({
-      requestId: 'id-6',
-      method: 'GET',
-      url: 'http://example.com/quux.css',
-      headers: {},
-      resourceType: 'fetch',
-      originalResourceType: undefined,
-      initiator: {
-        type: 'script',
-        url: 'http://example.com/baz.css?foo=bar',
-      },
-      documentURL: 'http://example.com/index.html?foo=bar',
-      cdpRequestWillBeSentTimestamp: 0,
-      cdpRequestWillBeSentReceivedTimestamp: 0,
-    })).to.be.false
-
-    // A preload request is not controlled
-    expect(manager.processBrowserPreRequest({
-      requestId: 'id-7',
-      method: 'GET',
-      url: 'http://example.com/quux.css',
-      headers: {},
-      resourceType: 'fetch',
-      originalResourceType: undefined,
-      initiator: {
-        type: 'preload',
-      },
-      documentURL: 'http://localhost:8080/index.html?foo=bar',
-      cdpRequestWillBeSentTimestamp: 0,
-      cdpRequestWillBeSentReceivedTimestamp: 0,
-    })).to.be.false
-  })
+        })
+
+        manager.addInitiatorToServiceWorker({
+          scriptURL: 'http://localhost:8080/sw.js',
+          initiatorOrigin: 'http://localhost:8080/',
+        })
+
+        // A script request emanated from the service worker's initiator is controlled
```

Comment:
Comments in this `it` do not match the assertions

Reply:
Updated comments [61cc17a](https://github.com/cypress-io/cypress/pull/28900/commits/61cc17a9442aebc5ae9dc7c774a5650c213e4150)